sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SpecificMutableRow.scala

```diff
@@ -202,6 +221,8 @@ final class SpecificMutableRow(val values: Array[MutableValue]) extends MutableR
         case DoubleType => new MutableDouble
         case BooleanType => new MutableBoolean
         case LongType => new MutableLong
+        // TODO(davies): enable this
+        // case StringType => new MutableString
```

Comment:
We need more thinking about reuse the UTF8String object, it's not a trivial decision, so I'd like to leave this out of this PR.